codility_training/lessons.lesson10.PrimeAndCompositeNumbers.Flags/sAp00n.py

```diff
@@ -0,0 +1,49 @@
+#  [3, 2, 1]
+#[0, 0, 0, 0, 0, 1, 0, 1, 0, 1]
+#[1,5,3,4,3,4,1,2,3,4,6,2]
+def solution(A):
+    list_of_peaks = find_peaks(A)
+    print(f'list_of_peaks= {list_of_peaks}')
+    number_of_peaks = len(list_of_peaks)
+
+    distance_of_peaks = []
+    if number_of_peaks > 1:
+
+        fst_peak = list_of_peaks[0]
+        for peak_idx in list_of_peaks[1:]:
+            currnet_peak = peak_idx
+            distance_of_peaks.append(currnet_peak - fst_peak)
+        print(f'distance_of_peaks= {distance_of_peaks}')
+
+        max_num_of_flag = 1
+        for i in range(len(distance_of_peaks) + 1, 1, -1):
+            #print(f'i= {i}')
+            temp = 1
+            for j in range(len(distance_of_peaks)):
```

Comment:
```suggestion
        peaks_len = len(distance_of_peaks)
        for i in range(peaks_len + 1, 1, -1):
            #print(f'i= {i}')
            temp = 1
            for j in range(peaks_len):
```
len 의 경우 생각보다 비용이 큽니다.